lib/index.ml

```diff
@@ -225,9 +232,23 @@ let active_accounts = ref Account_set.empty
 let set_active_accounts x = active_accounts := x
 let get_active_accounts () = !active_accounts
 
-let active_refs = ref Repo_map.empty
+let active_refs : (string * string) list Repo_map.t ref = ref Repo_map.empty
 
 let set_active_refs ~repo (refs : (string * string) list) =
+  (* Remove statuses from Status_cache corresponding to removed refs *)
+  Repo_map.find_opt repo !active_refs
+  |> Option.iter (fun old_refs ->
+    (* Set difference: find refs that have been removed by merging *)
+    let removed_refs =
+      List.filter
+        (fun k -> not @@ List.exists ((=) k) refs)
```

Comment:
nitpick: This could potentially also become 

```ocaml
(fun k -> not @@ List.mem k refs)
```

Reply:
Great, thanks! I was sure that function existed under some name or another...